neon_mq_connector/consumers/select_consumer.py

```diff
@@ -207,8 +207,8 @@ def run(self):
                 self.connection: pika.SelectConnection = self.create_connection()
                 self.connection.ioloop.start()
             except Exception as e:
-                LOG.error(f"Failed to start io loop on consumer thread {self.name!r}: {e}")
-                self._close_connection()
+                LOG.error(f"Failed to start io loop on consumer thread {self.name!r}: {e}, reconnecting")
+                self.reconnect()
```

Comment:
This will recursively call `run`. This can lead to a Maximum Recursion Error after some amount of time.